pandas/io/common.py

```diff
@@ -233,20 +199,16 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
     -------
     a filepath_or_buffer, the encoding, the compression
     """
-
+    
     if _is_url(filepath_or_buffer):
-        req = _urlopen(str(filepath_or_buffer))
-        if compression == 'infer':
-            content_encoding = req.headers.get('Content-Encoding', None)
-            if content_encoding == 'gzip':
-                compression = 'gzip'
-            else:
-                compression = None
-        # cat on the compression to the tuple returned by the function
-        to_return = (list(maybe_read_encoded_stream(req, encoding,
-                                                    compression)) +
-                     [compression])
-        return tuple(to_return)
+        url = str(filepath_or_buffer)
+        req = _urlopen(url)
+        content_encoding = req.headers.get('Content-Encoding', None)
+        if content_encoding == 'gzip':
+            # Override compression based on Content-Encoding header
+            compression = 'gzip'
```

Comment:
> i think inferring based on the file name is usually enough - but u can also use the Content if it's available

@jreback, `Content-Encoding` is not a versatile way to infer compression -- `gzip` is the only compression encoded that pandas and the Content-Encoding specification both support. I've modified it so inference on a URL uses its extension. However, if `Content-Encoding` is `gzip`, then compression is overridden. I don't expect this situation to arise very often. Does this make sense or should we just totally ignore `Content-Encoding`?